cryostatio/cryostat3
# chore(http): remove Apache HttpClient5, replace with RESTEasy client

# Welcome to Cryostat3! 👋
## Before contributing, make sure you have:

**To recreate commits with GPG signature** `git fetch upstream && git rebase --force --gpg-sign upstream/main`
_______________________________________________

Fixes: #382
Depends on #342
Depends on #368

## Description of the change:
- removes Apache HttpClient5 dependency
- replaces HttpClient5 utilities with utilities from other dependencies (HTTP content type and response status code constants)
- replaces HttpClient5 actual client instance for sidecar report generation with RESTEasy client, which is *The Quarkus Way* and already a dependency

## Motivation for the change:
One less dependency to manage, more idiomatic.

## How to manually test:
1. Build PR (currently also needs manual merge of #368)
2. `./smoketest.bash -OGr -s localstack`
3. In another terminal, `podman logs -f compose_reports_1`
4. Open web client UI, generate a report
5. Check reports container log, verify that it received and processed the request (and therefore that the new HTTP client was used to send it the data)
